src/CashuWallet.ts

```diff
@@ -63,7 +64,14 @@ class CashuWallet {
 		return {
 			isPaid: payData.paid ?? false,
 			preimage: payData.preimage,
-			change: payData?.change ? dhke.constructProofs(payData.change, rs, secrets, this.keys) : []
+			change: payData?.change
+				? dhke.constructProofs(
+						payData.change,
+						rs,
+						secrets,
+						payData.change.length ? await this.getKeys(payData.change[0].id) : []
+				  )
+				: []
```

Comment:
i wonder if it would make sense to return something that indicates if a keyset has changed. We're already returning a bunch of params, what's one more, eh?

Reply:
should we return just a bool like keysChanged or the keys?

Reply:
i think returning the keys themselves would make the most sense.